src/main/scala/viper/silver/ast/utility/Functions.scala

```diff
@@ -223,17 +200,18 @@ object Functions {
     *         formed cycles involves the set of functions `fs`.
     */
   def findFunctionCyclesVia(program: Program, via: Function => Seq[Exp], subs: Function => Seq[Exp] = allSubexpressions)
```

Comment:
To enable the transformation I mentioned above, I guess we could transform this into a func with type `def findFunctionCyclesVia(program: Program, via: FuncName => Seq[Exp], subs: FuncName => Seq[Exp])`.

Here, we would need a new default value to `subs`. I guess we could introduce a function that, given a `FuncName`, we first would need to find the corresponding function in `program.functions` using `find`.
I guess we would need to figure out if the cost incurred by searching in the list of functions when we fall in the second case is less so than the cost of having `findFunctionCyclesViaPreconditions` returning a `Map[Function, Set[Function]]`